lib/worker.js

```diff
@@ -167,7 +168,20 @@ var HTTPListner = function() {
 				response.end();
 			}
 		},
+		'/analyze-url' : function( request, response ) {
+			response.writeHead( 200, { 'Content-Type': 'text/plain' } );
+			response.write("Analyze");
+			response.end();
 
+			const url = new URL( request.url, "http://example.com/" );
```

Comment:
Setting the base url means that if `request.url` were to be "foo", it would analyze `http://example.com/foo`.

If the intention is just to set a default, I might just do `new URL( request.url ?  request.url : "http://example.com/")`

Reply:
The host was actually irrelevant. The second parameter is to set a base that is required when passing in a relative URL as the first parameter.  The `request.url` contains a relative URL which is why a base was set. It did not really matter what the base was, I was purely interested in the GET params. I've changed this code a bit to make the intention a bit clearer.

Reply:
ahh I see, ok so it just needs to be set to anything so you extract the params, ok fair enough 👍